.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.3.0
+  hooks:
+  - id: check-added-large-files
+  - id: end-of-file-fixer
+    files: 'repository_service_tuf/'
+  - id: trailing-whitespace
+    files: 'repository_service_tuf/'
+  - id: check-yaml
+    files: '.github/'
+- repo: https://github.com/pycqa/flake8
```

Comment:
Is it required to use the `repo` with version (`rev`) here? Once we already have our Pipenv/requirements-dev?

Reply:
Unfortunately, from what I researched it is required (rev is also a required key). requirements-type files and pre-commit-config.yaml don't cooperate together, see https://github.com/pre-commit/pre-commit/issues/730 and https://github.com/pre-commit/pre-commit/issues/945 (accordingly for JS).

There are different mitigation to this:
1) removing these linters from Pipenv/requirements-dev and using [pre-commit autoupdate](https://pre-commit.com/#updating-hooks-automatically) or manual update for the versions (cons: another place to keep track of versions)

2) Running tox from pre-commit as a local hook (see what I did with [tox-requirements hook](https://github.com/vmware/repository-service-tuf-cli/pull/110/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R27)). This way we are installing from tox the versions specified in Pipenv/requirements-dev
cons: we will not be able to take advantage of a key feature of pre-commit; that you can run pre-commit only on the last commit (here I use for the first time pre-commit run --all-files). This is extremely fast and also gives you the chance to slowly migrate changes if you are adapting to another linting format or adding something totally new at some point in the future, without having to refactor the whole code at once for the linter check

Reply:
I think the tool is very nice. I like the idea.

The two points that I'm not fully convinced about are the usage and dependencies on versioning maintenance.
- About the usage: The developer/contributor still need to install (`pre-commit install`) it by himself, having a big chance to them skip it. I say that because many contributors don't run `tox -r`. 
- About the dependency: `.pre-commit-config.yaml`` has its dependency version list, and [it doesn't work with dependabot.](https://github.com/dependabot/dependabot-core/issues/1524)

I would like to hear from @MVrachev and @rdimitrov 😄